src/statmodels.jl

```diff
@@ -76,6 +76,24 @@ This is usually the model containing only the intercept.
 nullloglikelihood(model::StatisticalModel) =
     error("nullloglikelihood is not defined for $(typeof(model)).")
 
+"""
+    pointwiseloglikelihood(model::StatisticalModel)
+
+Return a vector of each observation's contribution to the log-likelihood of the model.
+
+In general, `sum(pointwiseloglikehood(model)) == loglikelihood(model)`.
+"""
+pointwiseloglikelihood(model::StatisticalModel) =
+    error("pointwiseloglikelihood(model::StatisticalModel) is not defined for $(typeof(model)).")
+
+"""
+    pointwiseloglikelihood(model::StatisticalModel, observation)
+
+Return the log-likelihood of `observation` contribution under `model`.
+"""
+pointwiseloglikelihood(model::StatisticalModel, observation) =
```

Comment:
Do you expect observation to be anything other than integer?

Reply:
Yes -- I don't expect it to be simply an index, it could be e.g. a vector corresponding to a row in the model matrix in a linear model.`